core/com.b2international.snowowl.api.rest/src/com/b2international/snowowl/api/rest/validation/ValidationRestService.java

```diff
@@ -0,0 +1,321 @@
+					.setRequest(request)
+					.setDescription(String.format("Validating SNOMED CT on branch '%s'", validationInput.branchPath()))
+					.setUser(principal.getName())
+					.setId(uniqueJobId)
+					.buildAsync()
+					.execute(bus);
+			})
+			.then(success -> {
+				final String validationUuid = UUID.randomUUID().toString();
+				uniqueIdByUuid.put(validationUuid, uniqueJobId);
+				
+				final URI responseURI = linkBuilder.slash(validationUuid).toUri();
+				return Responses.created(responseURI).build();
+				
+			})
+			.fail(e -> {
+				return Responses.status(HttpStatus.BAD_REQUEST).build();
+			}));
+	}
+	
+	@ApiOperation(
+			value="Retrieve the state of a validation run from branch")
+	@ApiResponses({
+		@ApiResponse(code = 200, message = "OK"),
+		@ApiResponse(code = 404, message = "Validation not found", response=RestApiError.class)
+	})
+	@RequestMapping(value="/validations/{validationId}", method=RequestMethod.GET)
+	public @ResponseBody DeferredResult<RemoteJobEntry> getValidationRun(
+			@ApiParam(value="The validation identifier")
+			@PathVariable(value="validationId") 
+			final String validationId) {
+		
+		final String uniqueJobId = uniqueIdByUuid.get(validationId);
+		if (uniqueJobId != null) {
+			return DeferredResults.wrap(JobRequests.prepareGet(uniqueJobId)
+					.buildAsync()
+					.execute(bus));
+			
+		} else {
+			throw new NotFoundException("Validation job", validationId);
+		}
+				
+	}
+	
+	@ApiOperation(
+			value="Retrieve the validation issues from a completed validation on a branch.")
+	@ApiResponses({
+		@ApiResponse(code = 200, message = "OK"),
+		@ApiResponse(code = 404, message = "Branch not found", response=RestApiError.class)
+	})
+	@RequestMapping(value="/validations/{branchPath}/issues", method=RequestMethod.GET)
+	public @ResponseBody DeferredResult<ValidationIssues> getValidationResults(
+			@ApiParam(value="Branch path where the validation was run on.")
+			@PathVariable(value="branchPath")
+			final String branchPath,
+			
+			@ApiParam(value="The maximum number of items to return")
+			@RequestParam(value="limit", defaultValue="50", required=false)   
+			final int limit) {
+		
+		return DeferredResults.wrap(ValidationRequests.issues().prepareSearch()
+			.all()
+			.filterByBranchPath(branchPath)
+			.setLimit(limit)
+			.buildAsync()
+			.execute(bus));
+	}
+	
+	@ApiOperation(
+			value="Retrieve the validation issues from a completed validation on a branch.")
+	@ApiResponses({
+		@ApiResponse(code = 200, message = "OK"),
+		@ApiResponse(code = 404, message = "Branch or validation not found", response=RestApiError.class)
+	})
+	@RequestMapping(value="/validations/{branchPath}/issues", method=RequestMethod.GET, produces={AbstractRestService.CSV_MEDIA_TYPE})
+	public @ResponseBody DeferredResult<Collection<Object>> getValidationResultsAsCsv(
+			@ApiParam(value="Branch path where the validation was run on.")
+			@PathVariable(value="branchPath")
+			final String branchPath,
```

Comment:
If you are using `branchPath` in the API, then please:
1. Name it `path` just the other APIs do
2. Specify proper Ant-style wild card path matching to match deep branch paths (`/path:**`)

Also, I would still use the unique validation ID here instead of a branch path. You can get the branchPath from the parameters of the Validation Remote Job Entry's.